### PR TITLE
MockServer IMPS, actionManager Tests, navBar currentApp

### PR DESCRIPTION
**Pull request by kebeclibre**

### CODE REVIEW

File: addons/wowl/static/src/components/control_panel/control_panel.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class ControlPanel extends Component {
   }
 
   // Demo code
-  _onExecuteAction(actionId: number) {
-    this.actionManager.doAction(actionId);
+  _onExecuteAction(action: any) {
```
Comment (aab-odoo) [inline addons/wowl/static/src/components/control_panel/control_panel.ts]: we have a type for this (`ActionRequest`) :-)

File: addons/wowl/static/src/components/navbar/navbar.ts [hunk 2]
```diff
@@ -9,13 +9,21 @@ export class NavBar extends Component<{}, OdooEnv> {
   state = useState({ showDropdownMenu: false });
 
   systrayItems = this.env.registries.systray.getAll();
+  constructor(...args: any[]) {
+    super(...args);
+    this.env.bus.on("menus:app-changed", this, () => this.render());
```
Comment (aab-odoo) [inline addons/wowl/static/src/components/navbar/navbar.ts]: FYI, I changed `action_manager:update` to `ACTION_MANAGER:UPDATE` as it seems that there was some sort of convention to write (main bus) events in uppercase

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 3]
```diff
@@ -25,6 +26,7 @@ export interface ActionDescription {
 type ActionRequest = ActionId | ActionXMLId | ActionTag | ActionDescription;
 interface ActionOptions {
   clearBreadcrumbs?: boolean;
+  _menuId?: Menu['id'];
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: why prefixed by _?

Comment (aab-odoo) [reply addons/wowl/static/src/services/action_manager/action_manager.ts]: also, not sure (yet) this should be a doAction option

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 4]
```diff
@@ -285,6 +291,7 @@ function makeActionManager(env: OdooEnv): ActionManager {
       Component = controller.Component;
       mounted() {
         controllerStack = nextStack; // the controller is mounted, commit the new stack
+        env.services.menus.setCurrentMenu(action._menuId);
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: what about triggering an event on the bus?

File: addons/wowl/static/tests/helpers/mocks.ts [hunk 5]
```diff
@@ -80,29 +79,33 @@ export function makeFakeMenusService(menuData?: MenuData): Service<MenuService>
       return menusService;
     },
   };
+}*/
+
+function buildMockRPC(mockRPCs?: MockRPC | MockRPC[]) {
+  return async (...args: Parameters<RPC>) => {
+    let res;
+    if (mockRPCs) {
+      if (Array.isArray(mockRPCs)) {
```
Comment (aab-odoo) [inline addons/wowl/static/tests/helpers/mocks.ts]: do you have a usecase where mockRPCs is an array?

**Review by aab-odoo — COMMENTED** [review]
Cool PR, not sure about the way to update the current menu tough. We should talk about that

File: addons/wowl/static/tests/helpers/mocks.ts (hunk 5 above)
Comment (kebeclibre) [reply addons/wowl/static/tests/helpers/mocks.ts]: not really, The previous implementation had a global array to make sure other modules could push in it and cover default routes. Not sure if we will need it later on....

File: addons/wowl/static/src/services/action_manager/action_manager.ts (hunk 4 above)
Comment (kebeclibre) [reply addons/wowl/static/src/services/action_manager/action_manager.ts]: yes, i'm not sure either on how to update the navbar....

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**